### PR TITLE
Provider the APIs for the GitRepository custom resource

### PR DESCRIPTION
**Pull request by LinuxSuRen**
### What type of PR is this?

/kind feature

### What this PR does / why we need it:
As I mentioned in the title, this PR will add the APIs of the custom resource GitRepository.

And I change the version of `GitRepository` and `Addon` to `v1alpha3`. Because it belongs to the core part. Currently, all the core part are using `v1alpha3`.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:

Please check the following list before waiting reviewers:

- [ ] Already committed the CRD files to [the Helm Chart](https://github.com/kubesphere-sigs/ks-devops-helm-chart/) if you created some new CRDs
- [ ] Already [added the permission](https://github.com/kubesphere/ks-installer/blob/9e063b085a0e43fdb3d0d9e3e7f4149146f14b9c/roles/ks-core/prepare/files/ks-init/role-templates.yaml) for the new API
- [ ] Already added the RBAC markers for the new controllers

### Does this PR introduce a user-facing change??

```release-note
None
```

### CODE REVIEW

File: pkg/kapis/devops/v1alpha3/scm/scmhandler.go [hunk 1]
```diff
@@ -29,11 +29,15 @@ import (
 // handler holds all the API handlers of SCM
 type handler struct {
 	k8sClient client.Client
+	client.Client
```
Comment (JohnNiang) [inline pkg/kapis/devops/v1alpha3/scm/scmhandler.go:32]: IMO, `k8sClient client.Client` and `client.Client` are the same type. How about removing one of them?

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: pkg/kapis/devops/v1alpha3/scm/gitrepository_handler.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+	err := req.ReadEntity(patchRepo)
+	if err == nil {
+		err = h.Get(ctx, types.NamespacedName{
+			Namespace: namespace,
+			Name:      repoName,
+		}, repo)
+		if err == nil {
+			repo.Spec = patchRepo.Spec
+			err = h.Update(ctx, repo)
```
Comment (JohnNiang) [inline pkg/kapis/devops/v1alpha3/scm/gitrepository_handler.go:76]: Do we really update spec only? What about others? If it's true, it is recommended to mention it in the API doc.

BTW, This doubt won't prevent others from merging this PR.

Comment (LinuxSuRen) [reply pkg/kapis/devops/v1alpha3/scm/gitrepository_handler.go:76]: I'm not sure about that. Consider there is no UI that allows us to test it, I'm not able to think up enough details on that. I prefer to update the API once more details become clear.

**Review by JohnNiang — COMMENTED** [review]
/lgtm

**Pull request merged**